apps/src/applab/commands.js

```diff
@@ -1578,16 +1581,40 @@ function filterUrl(urlToCheck) {
   })
     .success(data => {
       let approved = data['approved'];
-      console.log('Url determined safe to open: ' + approved);
+      getStore().dispatch(actions.addRedirectNotice(approved, urlToCheck));
     })
     .fail((jqXhr, status) => {
-      console.log('Error. Please re-run program');
+      // When this query fails, default to the dialog that allows the user to choose
+      getStore().dispatch(actions.addRedirectNotice(true, urlToCheck));
     });
 }
 
 applabCommands.openUrl = function(opts) {
-  apiValidateType(opts, 'openUrl', 'url', opts.url, 'string');
-  filterUrl(opts.url);
+  if (apiValidateType(opts, 'openUrl', 'url', opts.url, 'string')) {
+    // Remove protocol from url string if present
+    let hostname = opts.url;
+    let protocols = ['https://', 'http://', 'www.'];
+    protocols.forEach(protocol => {
+      if (hostname.startsWith(protocol)) {
+        hostname = hostname.slice(protocol.length);
+      }
+    });
```

Comment:
nit: Since this section is only used for evaluating whether the hostname is studio/code.org, it would be clearer to group it under the studio/code.org comment you have below.